sys/dev/intpm/intpm.c

```diff
@@ -135,28 +136,41 @@ sb8xx_attach(device_t dev)
 	struct resource		*res;
 	uint32_t		devid;
 	uint8_t			revid;
-	uint16_t		addr;
+	uint32_t		addr = AMDSB_PMIO_INDEX;
 	int			rid;
 	int			rc;
 	bool			enabled;
 
 	sc = device_get_softc(dev);
+	devid = pci_get_devid(dev);
+	revid = pci_get_revid(dev);
+
+	/*
+	 * Comment from Linux i2c-piix4.c:
+	 *
+	 * cd6h/cd7h port I/O accesses can be disabled on AMD processors
+	 * w/ SMBus PCI revision ID 0x51 or greater. MMIO is supported on
+	 * the same processors and is the recommended access method.
+	 */
+	if (devid == AMDCZ_SMBUS_DEVID && revid >= AMDCZ51_SMBUS_REVID) {
+		sc->type = SYS_RES_MEMORY;
+		addr = AMDFCH41_MMIO_ADDR + AMDFCH41_MMIO_PM_OFF;
+	}
```

Comment:
I would set `sc_type` and `addr` in an else clause here as I think that will be clearer to future readers.

Reply:
Well, I guess I see why `sc->type` has to be set where it is, I would still move the `addr` into an `else` here.

Reply:
Good point. While there I included `sc->type = SYS_RES_IOPORT` in the else for clarity even though it is set by default to work in non sb8xx cases.